website/docs/d/resource.html.markdown

```diff
@@ -0,0 +1,44 @@
+---
+layout: "kubernetes"
+page_title: "Kubernetes: kubernetes_resource"
+description: |-
+  This is a generic data source for Kubernetes API resources
+---
+
+# kubernetes_resource
+
+This data source is a generic way to retrieve resources from the Kubernetes API. 
+
+### Example: Get data from a ConfigMap
+
+```hcl
+data "kubernetes_resource" "example" {
```

Comment:
@alexsomesan open to thoughts on the naming here, I didn't use `manifest` because that specifically refers to the YAML file you apply to the cluster whereas here we're retrieving an API resource so I thought `resource` made sense. `kubernetes_object` would also work too.

Reply:
I'm with you. This is a good choice of name. Using manifest would likely be more confusing. Also, this datasource could be used together with the native resources (non kubernetes_manifest) and then it makes even more sense to call it like this.